net/strongswan/files/ipsec.init

```diff
@@ -1,20 +1,327 @@
 #!/bin/sh /etc/rc.common
 
-START=90
-STOP=10
+NAME=ipsec
+START=21
+STOP=60
+USE_PROCD=1
 
-start() {
-	ipsec start
+. $IPKG_INSTROOT/lib/functions.sh
+. $IPKG_INSTROOT/lib/functions/service.sh
+. $IPKG_INSTROOT/lib/functions/network.sh
+
+FileSecrets=/var/ipsec/ipsec.secrets
+FileConn=/var/ipsec/ipsec.conf
+FileCommon=/var/ipsec/strongswan.conf
+
+EnabledRemotePeers=0
+
+ConfigPhase1() {
+	local encryption_algorithm
+	local hash_algorithm
+	local dh_group
+
+	config_get encryption_algorithm  "$1" encryption_algorithm
+	config_get hash_algorithm        "$1" hash_algorithm
+	config_get dh_group              "$1" dh_group
+
+	Phase1Proposal=${Phase1Proposal}","${encryption_algorithm}-${hash_algorithm}-${dh_group}
```

Comment:
Why suddenly switch to CamelCase? Coding style should be consistent. Good examples are dnsmasq (or really any init script in LEDE/OpenWrt base), or the one frome snmpd.